core/src/main/java/hudson/UDPBroadcastThread.java

```diff
@@ -103,10 +103,13 @@ public void run() {
             }
         } catch (ClosedByInterruptException e) {
             // shut down
-        } catch (BindException e) {
-            // if we failed to listen to UDP, just silently abandon it, as a stack trace
+        } catch (SocketException e) {
+            if (shutdown) { // forcibly closed
+                return;
+            }            // if we failed to listen to UDP, just silently abandon it, as a stack trace
             // makes people unnecessarily concerned, for a feature that currently does no good.
-            LOGGER.log(Level.WARNING, "Failed to listen to UDP port "+PORT,e);
+            LOGGER.log(Level.INFO, "Cannot listen to UDP port {0}, skipping: {1}", new Object[] {PORT, e});
+            LOGGER.log(Level.FINE, null, e);
```

Comment:
You should always use a message - as there is zero guarantee that the exception line will be printed immediately after the previous entry leading to exceptions that are "free floating" in the logs.